eth/accountmanager.go

```diff
@@ -147,10 +146,23 @@ func (am *accountManager) SignTx(signer types.Signer, tx *types.Transaction) (*t
 
 // Sign byte array message. Account must be unlocked
 func (am *accountManager) Sign(msg []byte) ([]byte, error) {
-	personalMsg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", 32, msg)
-	personalHash := crypto.Keccak256([]byte(personalMsg))
+	ethHash := accounts.TextHash(msg)
+	sig, err := am.keyStore.SignHash(am.account, ethHash)
+	if err != nil {
+		return nil, err
+	}
+
+	// sig is in the [R || S || V] format where V is 0 or 1
+	// Convert the V param to 27 or 28
+	v := sig[64]
+	if v == byte(0) {
```

Comment:
I think we can do something like `v = byte(int(v) + 27)` instead. Haven't tested it.

Reply:
Fixed in b309607